core/src/main/java/org/infinispan/statetransfer/StateTransferManagerImpl.java

```diff
@@ -98,9 +103,11 @@ public void start() throws Exception {
          log.tracef("Starting StateTransferManager of cache %s on node %s", cacheName, rpcManager.getAddress());
       }
 
+      Hash hashFunction = configuration.clustering().hash().hash();
+      componentRegistry.registerComponent(hashFunction, Hash.class);
```

Comment:
I meant a `HashFunctionFactory` as a component factory that does the wiring for the `Hash` instance configured by the user. There's no way to configure a `TransportFactory` either, it's just there so that the `Transport` instance it returns is properly wired with its dependencies.